e2e/test/FileUploadE2ETests.cs

```diff
@@ -54,6 +59,90 @@ public async Task FileUpload_X509_SmallFile_Http()
             await UploadFile(Client.TransportType.Http1, smallFile, true).ConfigureAwait(false);
         }
 
+        [TestMethod]
+        [TestCategory("LongRunning")]
+        public async Task FileUpload_SmallFile_Http_GranularSteps()
+        {
+            string filename = await GetTestFileNameAsync(FileSizeSmall).ConfigureAwait(false);
+            using FileStream fileStreamSource = new FileStream(filename, FileMode.Open, FileAccess.Read);
+            Http1TransportSettings fileUploadTransportSettings = new Http1TransportSettings();
+
+            await UploadFileGranular(fileStreamSource, filename, fileUploadTransportSettings).ConfigureAwait(false);
+        }
+
+        [TestMethod]
+        [TestCategory("LongRunning")]
+        public async Task FileUpload_SmallFile_Http_GranularSteps_x509()
+        {
```

Comment:
I don't see a sample (update or add) to this new set of API's. Is it another PR?

Reply:
The sample repo is separate for C#. I'll add a sample after this gets checked in